week1/citibike.R

```diff
@@ -1,48 +1,115 @@
-########################################
-
-# read one month of data
-trips <- read_csv('201402-citibike-tripdata.csv')
-
-# replace spaces in column names with underscores
-names(trips) <- gsub(' ', '_', names(trips))
-
-# convert dates strings to dates
-# trips <- mutate(trips, starttime = mdy_hms(starttime), stoptime = mdy_hms(stoptime))
-
-# recode gender as a factor 0->"Unknown", 1->"Male", 2->"Female"
-trips <- mutate(trips, gender = factor(gender, levels=c(0,1,2), labels = c("Unknown","Male","Female")))
-
-
-########################################
-# YOUR SOLUTIONS BELOW
-########################################
-
-# count the number of trips (= rows in the data frame)
-
-# find the earliest and latest birth years (see help for max and min to deal with NAs)
-
-# use filter and grepl to find all trips that either start or end on broadway
-
-# do the same, but find all trips that both start and end on broadway
-
-# find all unique station names
-
-# count the number of trips by gender, the average trip time by gender, and the standard deviation in trip time by gender
-# do this all at once, by using summarize() with multiple arguments
-
-# find the 10 most frequent station-to-station trips
-
-# find the top 3 end stations for trips starting from each start station
-
-# find the top 3 most common station-to-station trips by gender
-
-# find the day with the most trips
-# tip: first add a column for year/month/day without time of day (use as.Date or floor_date from the lubridate package)
-
-# compute the average number of trips taken during each of the 24 hours of the day across the entire month
-# what time(s) of day tend to be peak hour(s)?
+library(tidyverse)
+library(lubridate)
+
+########################################
+# READ AND TRANSFORM THE DATA
+########################################
+
+# read one month of data
+trips <- read_csv('201402-citibike-tripdata.csv')
+
+# replace spaces in column names with underscores
+names(trips) <- gsub(' ', '_', names(trips))
+
+# convert dates strings to dates
+# trips <- mutate(trips, starttime = mdy_hms(starttime), stoptime = mdy_hms(stoptime))
+
+# recode gender as a factor 0->"Unknown", 1->"Male", 2->"Female"
+trips <- mutate(trips, gender = factor(gender, levels=c(0,1,2), labels = c("Unknown","Male","Female")))
+
+
+########################################
+# YOUR SOLUTIONS BELOW
+########################################
+
+# count the number of trips (= rows in the data frame)
+
+trips %>%
+  nrow()
+
+# find the earliest and latest birth years (see help for max and min to deal with NAs)
+
+trips %>% 
+  mutate(birth_year = as.numeric(birth_year)) %>%
+  summarize(earliest_birth_year = min(birth_year, na.rm = TRUE), latest_birth_year = max(birth_year, na.rm = TRUE)) %>%
+  View
+
+# use filter and grepl to find all trips that either start or end on broadway
+
+trips %>%
+  filter(grepl('Broadway & .*', start_station_name) | grepl('.* & Broadway', end_station_name)) %>%

```

Comment:
The `& .` part of regex is a little redundant. You can just look for the presence of "Broadway"